server/src/main/java/io/crate/execution/engine/collect/files/FileReadingIterator.java

```diff
@@ -396,13 +398,7 @@ private BufferedReader createBufferedReader(InputStream inputStream) throws IOEx
         Predicate<URI> moduloPredicate;
         boolean sharedStorage = Objects.requireNonNullElse(shared, fileInput.sharedStorageDefault());
         if (sharedStorage) {
-            moduloPredicate = input -> {
-                int hash = input.hashCode();
-                if (hash == Integer.MIN_VALUE) {
-                    hash = 0; // Math.abs(Integer.MIN_VALUE) == Integer.MIN_VALUE
-                }
-                return Math.abs(hash) % numReaders == readerNumber;
-            };
+            moduloPredicate = input -> moduloPredicateImpl.test(input, this.readerNumber, this.numReaders);
```

Comment:
Why the changes here?

Reply:
Ah I see it's used in the tests. Any reason to wrap it in a `TriPredicate` instead of creating a function with the `URI -> int -> int` signature directly?

Reply:
no special reason, fixed it. Thank you